stock_average_daily_sale/models/stock_average_daily_sale.py

```diff
@@ -0,0 +1,336 @@
+# Copyright 2021 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from psycopg2.errors import ObjectNotInPrerequisiteState
+from psycopg2.extensions import AsIs
+
+from odoo import _, api, fields, models
+
+from odoo.addons.stock_storage_type_putaway_abc.models.stock_location import (
+    ABC_SELECTION,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class StockAverageDailySale(models.Model):
+
+    _name = "stock.average.daily.sale"
+    _auto = False
+    _order = "abc_classification_level ASC, product_id ASC"
+    _description = "Average Daily Sale for Products"
+
+    abc_classification_profile_id = fields.Many2one(
+        comodel_name="abc.classification.profile",
+        required=True,
+        index=True,
+    )
+    abc_classification_level = fields.Selection(
+        selection=ABC_SELECTION, required=True, readonly=True, index=True
+    )
+    average_daily_sales_count = fields.Float(
+        required=True, digits="Product Unit of Measure"
+    )
+    average_qty_by_sale = fields.Float(
+        required=True, digits="Product Unit of Measure", help="Average Daily Sales Qty"
+    )
+    average_daily_qty = fields.Float(
+        digits="Product Unit of Measure",
+        required=True,
+        help="The average daily qty sold",
+    )
+    config_id = fields.Many2one(
+        string="Stock Average Daily Sale Configuration",
+        comodel_name="stock.average.daily.sale.config",
+        required=True,
+    )
+    date_from = fields.Date(string="From", required=True)
+    date_to = fields.Date(string="To", required=True)
+    is_mto = fields.Boolean(
+        string="On Order",
+        readonly=True,
+        store=True,
+        index=True,
+    )
+    nbr_sales = fields.Integer(string="Number of Sales", required=True)
+    product_id = fields.Many2one(
+        comodel_name="product.product", string="Product", required=True, index=True
+    )
+    safety = fields.Float(
+        required=True,
+        help="daily standard deviation * safety factor * sqrt(nbr days into period "
+        "without saturday and sunday",
+    )
+    safety_bin_min_qty = fields.Float(
+        required=True,
+        digits="Product Unit of Measure",
+        help="Minimal safety qty into a bin location computed as: "
+        "average daily qty * number days in stock * safety",
+    )
```

Comment:
It is **+** safety instead of ***** safety.
I would rename this as the **recommended_qty**
```suggestion
    recommended_qty = fields.Float(
        required=True,
        digits="Product Unit of Measure",
        help="Minimal recommended quantity in stock. Formula: average daily qty * number days in stock + safety",
    )
```